crates/rpc/tests/rpc.rs

```diff
@@ -1,192 +1,147 @@
-
-use cached::SizedCache;
 use reqwest::Url;
 use starknet::{
     core::types::{
         BlockId, BlockWithTxHashes, BlockWithTxs, DeclareTransaction,
         DeployAccountTransaction, FieldElement, InvokeTransaction,
-        MaybePendingBlockWithTxHashes, MaybePendingBlockWithTxs,
-        MaybePendingTransactionReceipt, Transaction, TransactionReceipt,
-    },
-    providers::{
-        jsonrpc::HttpTransport, JsonRpcClient, Provider, ProviderError,
+        MaybePendingBlockWithTxHashes, MaybePendingTransactionReceipt,
+        Transaction, TransactionReceipt,
     },
+    providers::{jsonrpc::HttpTransport, JsonRpcClient, Provider},
 };
 
-lazy_static::lazy_static! {
-    static ref BLOCK_CACHE: Mutex<SizedCache<u64, MaybePendingBlockWithTxs>> = Mutex::new(SizedCache::with_size(1000));
-}
+const TEST_RPC_URL: &str = "http://localhost:3030";
 
 fn rpc_client() -> JsonRpcClient<HttpTransport> {
-    let rpc_url: Url = format!("http://localhost:{}", DEFAULT_PORT)
-        .parse()
-        .expect("Invalid RPC URL");
-    JsonRpcClient::new(HttpTransport::new(rpc_url))
+    let url: Url = TEST_RPC_URL.try_into().expect("Invalid RPC URL");
+    JsonRpcClient::new(HttpTransport::new(url))
 }
 
-struct TestClient {
-    inner: JsonRpcClient<HttpTransport>,
-}
-
-impl TestClient {
-    async fn get_cached_block_with_txs(
-        &mut self,
-        number: u64,
-    ) -> Result<MaybePendingBlockWithTxs, ProviderError> {
-        let mut cache = BLOCK_CACHE.lock().expect("Poisoned lock");
-        match cache.cache_get(&number) {
-            Some(block) => Ok(block.clone()),
-            None => {
-                let block =
-                    self.get_block_with_txs(BlockId::Number(number)).await?;
-                cache.cache_set(number, block.clone());
-                Ok(block)
-            }
-        }
-    }
-}
+mod blocks {
+    use futures::{stream, Stream, StreamExt};
+    use starknet::{
+        core::types::{BlockId, BlockWithTxs, MaybePendingBlockWithTxs},
+        providers::{jsonrpc::HttpTransport, JsonRpcClient, Provider},
+    };
 
-impl Deref for TestClient {
-    type Target = JsonRpcClient<HttpTransport>;
+    use super::rpc_client;
 
-    fn deref(&self) -> &Self::Target {
-        &self.inner
+    struct State {
+        client: JsonRpcClient<HttpTransport>,
+        latest: Option<u64>,
     }
-}
 
-struct TestContext<T> {
-    client: TestClient,
-    block: BlockWithTxs,
-    block_id: BlockId,
-    extracted_value: T,
-}
-
-/// Creates a `TestContext` with the latest block.
-async fn latest_block_context() -> TestContext<()> {
-    context(|_block| Some(())).await
-}
-
-/// Creates a `TestContext` with the latest block with at least N transactions.
-async fn n_txs_context(min_tx_number: usize) -> TestContext<()> {
-    context(|block| {
-        if block.transactions.len() >= min_tx_number {
-            Some(())
+    async fn next(state: State) -> Option<(BlockWithTxs, State)> {
```

Comment:
```suggestion
    async fn next_state(state: State) -> Option<(BlockWithTxs, State)> {
```
nit: I was confused at first and thought that the function produces the next block because it is in the module `blocks::next`, but then I realized that it is supposed to produce the next state for the unfold, hence my name change suggestion 😄

Reply:
`change_stream_state` it is now